sources/Renderer/OpenGL/Platform/Linux/LinuxGLSwapChainContext.cpp

```diff
@@ -22,58 +26,123 @@ namespace LLGL
 
 std::unique_ptr<GLSwapChainContext> GLSwapChainContext::Create(GLContext& context, Surface& surface)
 {
-    return MakeUnique<LinuxGLSwapChainContext>(static_cast<LinuxGLContext&>(context), surface);
+#ifdef LLGL_LINUX_ENABLE_WAYLAND
+    if (surface.IsWayland()) {
+        return MakeUnique<LinuxWaylandGLSwapChainContext>(static_cast<LinuxGLContext&>(context), surface);
+    } else {
+        return MakeUnique<LinuxX11GLSwapChainContext>(static_cast<LinuxGLContext&>(context), surface);
+    }
+#else
+    return MakeUnique<LinuxX11GLSwapChainContext>(static_cast<LinuxGLContext&>(context), surface);
+#endif
 }
 
-bool GLSwapChainContext::MakeCurrentUnchecked(GLSwapChainContext* context)
+
+/*
+ * LinuxX11GLSwapChainContext class
+ */
+
+LinuxX11GLSwapChainContext::LinuxX11GLSwapChainContext(LinuxGLContext& context, Surface& surface) :
+    GLSwapChainContext { context                 },
+    glc_               { static_cast<GLXContext>(context.GetGLXContext()) }
 {
-    return LinuxGLSwapChainContext::MakeCurrentGLXContext(static_cast<LinuxGLSwapChainContext*>(context));
+    /* Get native window handle */
+    NativeHandle nativeHandle = {};
+    if (surface.GetNativeHandle(&nativeHandle, sizeof(nativeHandle)))
+    {
+        dpy_ = nativeHandle.x11.display;
+        wnd_ = nativeHandle.x11.window;
+    }
+    else
+        LLGL_TRAP("failed to get X11 Display and Window from swap-chain surface");
 }
 
+bool LinuxX11GLSwapChainContext::HasDrawable() const
+{
+    return (wnd_ != 0);
+}
+
+bool LinuxX11GLSwapChainContext::SwapBuffers()
+{
+    glXSwapBuffers(dpy_, wnd_);
+    return true;
+}
+
+void LinuxX11GLSwapChainContext::Resize(const Extent2D& resolution)
+{
+    // dummy
+}
+
+bool LinuxX11GLSwapChainContext::MakeCurrentUnchecked()
+{
+    return glXMakeCurrent(dpy_, wnd_, glc_);       
+}
+
+bool LinuxX11GLSwapChainContext::Destroy() {
```

Comment:
The idea behind all the `MakeContextCurrent` functions being static members is so that you can also pass a null pointer to handle the case of removing the current context from the internal GL state. This is intended to be independent of any LLGL owned context to be alive, so this refactoring of `MakeCurrentUnchecked()`/`Destroy()` being non-static member functions breaks this concept. In what way is this refactor necessary for Wayland support?